### ISSUE TITLE
columns order

### ISSUE THREAD

**Issue by samehif** [dmlc/xgboost#2414]
changing the order of columns of my input data changes the xgboost prediction. i am using python on Ubuntu and i fixed the seed.

my xgboost version is : 0.6a2

**Comment by khotilov** [comment]
The order of columns matters - it's a feature.

**Comment by samehif** [comment]
how it matters? can you explain more? is it because of the col sub sampling? because i even tried with 
`    'colsample_bytree': 1.0,
    'colsample_bylevel': 1.0,
`
and i still get different results. can you point you to anywhere in the documentation where they talk about feature order?

**Comment by khotilov** [comment]
Xgboost identifies columns by their index. Thus, it doesn't make sense to do predictions in new data with different column order than in training.

If your question is about different order during training, then if some columns happen to have the same loss reduction, then the column with the lowest index is picked up.

**Comment by samehif** [comment]
yes. my question is about columns order during training (my training and validation set always have the same order of columns between them). 

isn't better if you sort the columns before training to avoid this?
